openedx/core/djangoapps/content_staging/models.py

```diff
@@ -51,9 +57,9 @@ class Meta:
             e.g. "video" if a video is staged, or "vertical" for a unit.
         """),
     )
-    olx = models.TextField(null=False, blank=False)
+    olx = MultiCollationTextField(null=False, blank=False, db_collations=CASE_SENSITIVE_COLLATIONS)
     # The display name of whatever item is staged here, i.e. the root XBlock.
-    display_name = models.CharField(max_length=1024)
+    display_name = case_insensitive_char_field(max_length=1024)
```

Comment:
If we want to keep this indexable in `utf8mb4`, it has to come down to a max length of no more than 768 characters, per: https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-conversion.html

> For InnoDB tables that use [COMPRESSED](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_compressed_row_format) or [DYNAMIC](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_dynamic_row_format) row format, [index key prefixes](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_index_prefix) longer than 767 bytes (up to 3072 bytes) are permitted. Tables created with these row formats enable you to index a maximum of 1024 or 768 characters for utf8mb3 or utf8mb4 columns, respectively. For related information, see [Section 15.22, “InnoDB Limits”](https://dev.mysql.com/doc/refman/8.0/en/innodb-limits.html), and [DYNAMIC Row Format](https://dev.mysql.com/doc/refman/8.0/en/innodb-row-format.html#innodb-row-format-dynamic).